src/test/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtOpeningTimesControllerTest.java

```diff
@@ -0,0 +1,97 @@
+    private static final String BASE_PATH = "/courts/";
+    private static final String CHILD_PATH = "/openingTimes";
+    private static final String TEST_SLUG = "unknownSlug";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Autowired
+    private transient MockMvc mockMvc;
+
+    @MockBean
+    private AdminCourtOpeningTimeService adminService;
+
+    @Test
+    void retrieveOpeningTimesShouldReturnCourtOpeningTimes() throws Exception {
+        final String expectedJson = getOpeningTimesJson();
+        final List<OpeningTime> openingTimes = asList(OBJECT_MAPPER.readValue(expectedJson, OpeningTime[].class));
+
+        when(adminService.getCourtOpeningTimesBySlug(TEST_SLUG)).thenReturn(openingTimes);
+
+        mockMvc.perform(get(BASE_PATH + TEST_SLUG + CHILD_PATH))
+            .andExpect(status().isOk())
+            .andExpect(content().json(expectedJson));
+    }
+
+    @Test
+    void retrieveOpeningTimeShouldReturnNotFoundForUnknownCourtSlug() throws Exception {
+        when(adminService.getCourtOpeningTimesBySlug(TEST_SLUG)).thenThrow(new NotFoundException(TEST_SLUG));
+
+        mockMvc.perform(get(BASE_PATH + TEST_SLUG + CHILD_PATH))
+            .andExpect(status().isNotFound())
+            .andExpect(content().string("Not found: " + TEST_SLUG));
+    }
+
+    @Test
+    void updateOpeningTimesShouldReturnUpdatedCourtOpeningTimes() throws Exception {
+        final String expectedJson = getOpeningTimesJson();
+        final List<OpeningTime> openingTimes = asList(OBJECT_MAPPER.readValue(expectedJson, OpeningTime[].class));
+
+        when(adminService.updateCourtOpeningTimes(TEST_SLUG, openingTimes)).thenReturn(openingTimes);
+
+        mockMvc.perform(put(BASE_PATH + TEST_SLUG + CHILD_PATH)
+                            .content(expectedJson)
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(content().json(expectedJson));
+    }
+
+    @Test
+    void updateOpeningTimesShouldReturnNotFoundForUnknownCourtSlug() throws Exception {
+        final String json = getOpeningTimesJson();
+        final List<OpeningTime> openingTimes = asList(OBJECT_MAPPER.readValue(json, OpeningTime[].class));
+
+        when(adminService.updateCourtOpeningTimes(TEST_SLUG, openingTimes)).thenThrow(new NotFoundException(TEST_SLUG));
+
+        mockMvc.perform(put(BASE_PATH + TEST_SLUG + CHILD_PATH)
+                            .content(json)
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isNotFound())
+            .andExpect(content().string("Not found: " + TEST_SLUG));
+    }
+
+    private String getOpeningTimesJson() throws IOException {
+        final Path path = Paths.get("src/test/resources/opening-times.json");
```

Comment:
Should this be a variable up the top? I.e 

`TEST_OPENING_TIMES_PATH` ?

Reply:
Yes it can be for clarity